File: app/src/main/java/ru/esm/tspiot/dkkt/service/CashierDeviceInfoImpl.java
package ru.esm.tspiot.dkkt.service;

import android.os.Build;
import android.os.RemoteException;

public class CashierDeviceInfoImpl extends ICashierDeviceInfo.Stub {
    @Override
    public String getSerialNumber() throws RemoteException {
        return Build.SERIAL;
    }

    @Override
    public String getFirmwareVersion() throws RemoteException {
        return Build.VERSION.RELEASE;
    }

    @Override
    public String getConfigurationVersion() throws RemoteException {
        return Build.MANUFACTURER;
    }

    @Override
    public String getModelName() throws RemoteException {
        return Build.MODEL;
    }
}

File: app/src/kotlin/kotlin/ru/esm/tspiot/dkkt/service/CashierDeviceInfoImpl.kt
package ru.esm.tspiot.dkkt.service

import android.os.Build
import android.os.RemoteException

class CashierDeviceInfoImpl : ICashierDeviceInfo.Stub() {

    override fun getSerialNumber(): String = Build.SERIAL

    override fun getFirmwareVersion(): String = Build.VERSION.RELEASE
    override fun getConfigurationVersion(): String? = Build.VERSION.CODENAME

    override fun getModelName(): String? = Build.MODEL
}